### PR TITLE
Fix visual problem and add new style

### PR DESCRIPTION
**Pull request by MewX**
The bug shows like this in low resolution screen:

device-2015-01-23-203704

I fix it by drawing twice, which likes adding it a "thickness" attribute. It's the easiest way to fix, I guess because of the transition from "dp" to "px".
And I've tried to draw only once by adding 1 or 0.5f on "mHalf", problem still exists.

Here is the performance after drawing twice:
device-2015-01-23-203740

### CODE REVIEW

**Comment by MewX** [comment]
movie

Added this new style from "Chrome doodle-notifier".

**Comment by jpardogo** [comment]
Thank you very much for the PR. I am a bit busy at the moment. I will take a look asap.

File: example/src/main/java/com/jpardogo/android/googleprogressbar/SettingsActivity.java [hunk 1]
```diff
@@ -21,7 +21,7 @@ public static Intent newInstance(Context context){
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.settings);
-        getActionBar().setDisplayHomeAsUpEnabled(true);
+        // getActionBar().setDisplayHomeAsUpEnabled(true); // no actionbar on 4.0.3
```
Comment (jpardogo) [inline example/src/main/java/com/jpardogo/android/googleprogressbar/SettingsActivity.java]: Why not?

File: example/src/main/java/com/jpardogo/android/googleprogressbar/MainActivity.java [hunk 2]
```diff
@@ -19,11 +22,13 @@
...
 
 
-public class MainActivity extends Activity {
+public class MainActivity extends ActionBarActivity {
 
     private final int FOLDING_CIRCLES = 0;
     private final int MUSIC_DICES = 1;
     private final int NEXUS_CROSS_ROTATION = 2;
+    private final int CHROME_FLOATING_CIRCLES = 3;
+
```
Comment (jpardogo) [inline example/src/main/java/com/jpardogo/android/googleprogressbar/MainActivity.java]: Could we add static to this variables? I forgot as well I didn't remember.

File: example/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -7,18 +7,18 @@
...
-    <ProgressBar
-        android:id="@+id/google_progress"
-        android:layout_width="200dp"
-        android:layout_height="200dp"
-        android:layout_gravity="center"/>
+    <!--<ProgressBar-->
+        <!--android:id="@+id/google_progress"-->
+        <!--android:layout_width="200dp"-->
+        <!--android:layout_height="200dp"-->
+        <!--android:layout_gravity="center"/>-->
```
Comment (jpardogo) [inline example/src/main/res/layout/activity_main.xml]: The example work with share preference and the progressBar type is set dynamically. Use an standard progressBar then.

File: library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+
+        return;
+    }
+
+    @Override
+    protected void onBoundsChange(Rect bounds) {
+        super.onBoundsChange(bounds);
+        measureCircleProgress(bounds.width(), bounds.height());
+
+        return;
```
Comment (jpardogo) [inline library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java]: Unneccesary space

File: library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+        mPaint4.setAntiAlias(true);
+
+        return;
+    }
+
+    @Override
+    protected void onBoundsChange(Rect bounds) {
+        super.onBoundsChange(bounds);
+        measureCircleProgress(bounds.width(), bounds.height());
+
```
Comment (jpardogo) [inline library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java]: Unneccesary space

File: library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java [hunk 6]
```diff
@@ -0,0 +1,357 @@
...
+    @Override
+    protected boolean onLevelChange(int level) {
+
+        // calc one offset data is enough
+        // 0.5 * a * t^2 / mCenterPoint.x = level / sideLevel
+        // t from 0 to 10,000, so divided into 4 parts.
+        // the ACCELERATION_LEVEL defines how many divisions in 10000 levels
+        level %= MAX_LEVEL / acceleration;
+
```
Comment (jpardogo) [inline library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java]: Unneccesary space

File: library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java [hunk 7]
```diff
@@ -0,0 +1,357 @@
...
+                }
+
+                // from beg to mid
+                offsetPercentage = 0.5 * acceleration * temp_level * temp_level / distance;
+                offset = (int)(offsetPercentage * ef_width / 2); // x and y direction offset
+            }
+            else {
+                // set colorSign
+                colorSign |= 2;
+
```
Comment (jpardogo) [inline library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java]: Unneccesary space

File: library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java [hunk 8]
```diff
@@ -0,0 +1,357 @@
...
+                }
+
+                // from end to mid
+                offsetPercentage = 0.5 * acceleration * temp_level * temp_level  / distance;
+                offset = (int)(ef_width - offsetPercentage * ef_width / 2); // x and y direction offset
+            }
+            else {
+                // set colorSign
+                colorSign |= 8;
+
```
Comment (jpardogo) [inline library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java]: Unneccesary space

File: library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java [hunk 9]
```diff
@@ -0,0 +1,357 @@
...
+                // from mid to beg
+                offsetPercentage = ( max_speed * temp_level
+                        - 0.5 * acceleration * temp_level * temp_level ) / distance
+                        + 1.0;
+                offsetPercentage = offsetPercentage == 1.0 ? 2.0 : offsetPercentage;
+                offset = (int)(ef_width - offsetPercentage * ef_width / 2); // x and y direction offset
+
+
```
Comment (jpardogo) [inline library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java]: Unneccesary spaces

File: library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java [hunk 10]
```diff
@@ -0,0 +1,357 @@
...
+            }
+        }
+
+        mArrowPoints[0].set((int)unit+x_beg+offset, (int)unit+y_beg+offset); // mPaint1, left up
+        mArrowPoints[1].set((int)(unit*4.0)+x_beg-offset, (int)(unit*4.0)+y_beg-offset); // mPaint2, right down
+        mArrowPoints[2].set((int)unit+x_beg+offset, (int)(unit*4.0)+y_beg-offset); // mPaint3, left down
+        mArrowPoints[3].set((int)(unit*4.0)+x_beg-offset, (int)unit+y_beg+offset); // mPaint4, right up
+
```
Comment (jpardogo) [inline library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java]: Unneccesary space

File: library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java [hunk 11]
```diff
@@ -0,0 +1,357 @@
...
+                currentProgressStates = ProgressStates.GREEN_TOP;
+                break;
+        }
+
+        return;
+    }
+
+    @Override
+    public void draw(Canvas canvas) {
+
```
Comment (jpardogo) [inline library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java]: Unneccesary space

File: library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java [hunk 12]
```diff
@@ -0,0 +1,357 @@
...
+        mArrowPoints[3] = new Point((int)(unit*4.0)+x_beg, (int)unit+y_beg); // mPaint4, right up
+
+        return;
+    }
+
+    public void setAcceleration( int acceleration ) {
+        this.acceleration = acceleration;
+        distance = 0.5 * acceleration * ( MID_LEVEL / acceleration ) * ( MID_LEVEL / acceleration );
+        max_speed = acceleration * ( MID_LEVEL / acceleration );
+
```
Comment (jpardogo) [inline library/src/main/java/com/jpardogo/android/googleprogressbar/library/ChromeFloatingCirclesDrawable.java]: Unneccesary space

File: library/src/main/java/com/jpardogo/android/googleprogressbar/library/FoldingCirclesDrawable.java [hunk 13]
```diff
@@ -188,6 +188,10 @@ private void drawXMotion(Canvas canvas) {
         mPath.reset();
         mPath.moveTo(mHalf, 0);
         mPath.cubicTo(mAxisValue, 0, mAxisValue, mDiameter, mHalf, mDiameter);
+
```
Comment (jpardogo) [inline library/src/main/java/com/jpardogo/android/googleprogressbar/library/FoldingCirclesDrawable.java]: Unneccesary space

File: library/src/main/java/com/jpardogo/android/googleprogressbar/library/FoldingCirclesDrawable.java [hunk 14]
```diff
@@ -196,6 +200,10 @@ private void drawYMotion(Canvas canvas) {
         mPath.reset();
         mPath.moveTo(0, mHalf);
         mPath.cubicTo(0, mAxisValue, mDiameter, mAxisValue, mDiameter, mHalf);
+
```
Comment (jpardogo) [inline library/src/main/java/com/jpardogo/android/googleprogressbar/library/FoldingCirclesDrawable.java]: Unneccesary space

File: library/src/main/java/com/jpardogo/android/googleprogressbar/library/NexusRotationCrossDrawable.java [hunk 15]
```diff
@@ -55,6 +56,14 @@ private void initObjectAnimator() {
         final ObjectAnimator objectAnimator = ObjectAnimator.ofInt(this, "rotationAngle", 0, 180);
         objectAnimator.setInterpolator(LINEAR_INTERPOLATOR);
         objectAnimator.setDuration(ANIMATION_DURATION);
+
```
Comment (jpardogo) [inline library/src/main/java/com/jpardogo/android/googleprogressbar/library/NexusRotationCrossDrawable.java]: Unneccesary space

File: library/src/main/java/com/jpardogo/android/googleprogressbar/library/NexusRotationCrossDrawable.java [hunk 16]
```diff
@@ -55,6 +56,14 @@ private void initObjectAnimator() {
...
+        objectAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+            @Override
+            public void onAnimationUpdate(ValueAnimator animation) {
+                int v = (Integer)animation.getAnimatedValue();
+            }
+        });
+
```
Comment (jpardogo) [inline library/src/main/java/com/jpardogo/android/googleprogressbar/library/NexusRotationCrossDrawable.java]: Unneccesary addUpdateListener

**Comment by jpardogo** [comment]
Please, review my comments so we can continue and merge...

**Comment by MewX** [comment]
Plus, should use `getSupportActionBar()` instead of `getActionBar()`.
I fix this in the latest commit.

**Comment by jpardogo** [comment]
Merged into dev a1de603d3ea4a03a750eea15c4629c6489cc14bb

**Pull request closed without merging**